### PR TITLE
docs: Update licences and instances of DB Netz AG to DB InfraGO AG

### LINKED ISSUE #1225: Replace license headers with new entity
At the beginning of next year, the license headers should be replaced. 

Old license headers: 
```txt
# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
# SPDX-License-Identifier: Apache-2.0
```

New license headers:
```txt
# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
# SPDX-License-Identifier: Apache-2.0
```

### PR DESCRIPTION
**Pull request by romeonicholas**
Resolves #1225

### CODE REVIEW

**Comment by romeonicholas** [comment]
Hey @dominik003, it looks like because I'm touching every single file sonarcloud has an issue with existing concerns. I'm not sure I want to resolve those as Won't Fix without someone taking a look first though, if you're able  🐞

**Comment by dominik003** [comment]
> Hey @dominik003, it looks like because I'm touching every single file sonarcloud has an issue with existing concerns. I'm not sure I want to resolve those as Won't Fix without someone taking a look first though, if you're able 🐞

Sure, I'll look over! 

In addition, `mypy` seems to fail in our pre-commit hook with the following error:
```
Argument 2 to "add_exception_handler" of "Starlette" has incompatible type "Callable[[Request, UserNotFoundError], Coroutine[Any, Any, Response]]"; expected "Callable[[Request, Exception], Response | Awaitable[Response]] | Callable[[WebSocket, Exception], Awaitable[None]]"  [arg-type]
```

So essentially stating that our type:

```
Callable[[Request, UserNotFoundError], Coroutine[Any, Any, Response]]
```

is not one of the following expected one:

```
Callable[[Request, Exception], Response | Awaitable[Response]] | Callable[[WebSocket, Exception], Awaitable[None]]
```

However, as far as I can see, the our `UserNotFoundError` is a sub type of `Exception` and as stated in the [docs](https://docs.python.org/3/library/collections.abc.html#collections.abc.Coroutine), *All Coroutine instances are also instances of Awaitable*. So together with the fact that this just works fine, I would suggest to simply put `# type: ignore[arg-type]` to all of those lines.

The only weird thing is that this error just occurred now, so not quite sure what changed for it to occur.

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 A web platform for collaboration on MBSE and [Capella](https://www.eclipse.org/capella/) projects.
 
-**Copyright 2021 - 2023 [DB Netz AG](https://fahrweg.dbnetze.com/),
+**Copyright 2021 - 2023 [DB InfraGO AG](https://fahrweg.dbnetze.com/),
```
Comment (dominik003) [inline README.md:17]:
```suggestion
**Copyright 2021 - 2023 [DB InfraGO AG](https://fahrweg.dbinfrago.com/fahrweg-de),
```

**Review by dominik003 — APPROVED** [review]
Thank you for taking care of the renaming. There is only one link in the README, which we can already change to the new page. After that I can merge it.

File: README.md (hunk 1 above)
Comment (romeonicholas) [reply README.md:17]: Ooh good catch, though should I also update the copyright to 2024?

Comment (dominik003) [reply README.md]: Also good catch from you 😄 Feel free to also update the copyright year

Comment (romeonicholas) [reply README.md]: Not sure why I even asked, done 😆 ✅

**Comment by MoritzWeber0** [comment]
👍

**Pull request merged**